core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2073,8 +2073,9 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     // its own local file system, which is incorrect because the checkpoint files
     // are actually on the executor machines.
     if (!isLocal && Utils.nonLocalPaths(directory).isEmpty) {
-      logWarning("Checkpoint directory must be non-local " +
-        "if Spark is running on a cluster: " + directory)
+      logWarning(s"Spark is not running in local mode, therefore the checkpoint directory " +
```

Comment:
Interpolation here is on the wrong string of these 3. Also looks like some style check error